cornflow/endpoints/case.py

```diff
@@ -217,9 +247,63 @@ def get(self, idx):
         It requires authentication to be passed in the form of a token that has to be linked to
         an existing session (login) made by a user.
 
-        :param str idx: ID of the case
+        :param int idx: ID of the case
         :return: A dictionary with a message (error if authentication failed, or the execution does not exist or
           the data of the instance) and an integer with the HTTP status code.
         :rtype: Tuple(dict, integer)
         """
         return CaseModel.get_one_object_from_user(self.get_user(), idx)
+
+
+class CaseToInstance(MetaResource, MethodResource):
+    """
+    Endpoint used to create a new instance or instance and execution from a stored case
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.model = InstanceModel
+        self.query = InstanceModel.get_all_objects
+        self.primary_key = "id"
+
+    @doc(
+        description="Copies the information stored in a case into a new instance or instance and execution",
+        tags=["Cases"],
+    )
+    @Auth.auth_required
+    @marshal_with(CaseToLive)
+    def post(self, idx):
+        """
+        API method to copy the information stored in a case to a new instance or a new instance and execution.
+        It requires authentication to be passed in the form of a token that has to be linked to
+        an existing session (login) made by a user
+
+        :param int idx: ID of the case that has to be copied to an instance or instance and execution
+        :return: an object with the instance or instance and execution ID that have been created and the status code
+        :rtype: Tuple (dict, integer)
+        """
+        case = CaseModel.get_one_object_from_user(self.get_user(), idx)
+
+        if case is None:
+            raise ObjectDoesNotExist()
+
+        schema = case.schema
+        payload = {
+            "name": "instance_from_" + case.name,
+            "description": "Instance created from " + case.description,
+            "data": case.data,
+            "schema": case.schema,
+        }
+
+        if schema is None:
+            return self.post_list(payload)
+
+        if schema == "pulp" or schema == "solve_model_dag":
+            validate_and_continue(DataSchema(), payload["data"])
+            return self.post_list(payload)
+
+        config = current_app.config
+        marshmallow_obj = get_schema(config, schema)
+        validate_and_continue(marshmallow_obj(), payload["data"])
```

Comment:
we have this code in several places in the codebase (several endpoints). Maybe we should think on creating a single function that does this and call it from each. I'm not sure / convinced.

Reply:
I can think about it. Maybe it makes sense to generalize it if we also start to validate the config data for the executions.